src/transports/smart_pkt.c

```diff
@@ -407,17 +394,7 @@ int git_pkt_parse_line(
 
 	len = parse_len(line);
 	if (len < 0) {
-		/*
-		 * If we fail to parse the length, it might be because the
-		 * server is trying to send us the packfile already.
-		 */
-		if (bufflen >= 4 && !git__prefixcmp(line, "PACK")) {
-			giterr_clear();
-			*out = line;
-			return pack_pkt(head);
-		}
-
-		return (int)len;
+		return GIT_ERROR;
```

Comment:
We should set an error message here, perhaps:

    giterr_set(GITERR_NET, "unexpected pack file");
    return -1;

(We usually do `return -1` instead of `return GIT_ERROR` out of convention.  I'd like to keep that moving forward.)